src/mahoji/lib/abstracted_commands/blackjackCommand.ts

```diff
@@ -0,0 +1,211 @@
+import { channelIsSendable } from '@oldschoolgg/toolkit';
+import { ActionRowBuilder, ButtonBuilder, ButtonStyle, ChatInputCommandInteraction } from 'discord.js';
+import { Time } from 'e';
+import { CommandResponse } from 'mahoji/dist/lib/structures/ICommand';
+import { Bank } from 'oldschooljs';
+import { toKMB } from 'oldschooljs/dist/util';
+
+import { awaitMessageComponentInteraction } from '../../../lib/util';
+import { handleMahojiConfirmation } from '../../../lib/util/handleMahojiConfirmation';
+import { deferInteraction } from '../../../lib/util/interactionReply';
+import { mahojiParseNumber } from '../../mahojiSettings';
+
+const returnButtons = [
+	new ActionRowBuilder<ButtonBuilder>().addComponents([
+		new ButtonBuilder({
+			label: 'Hit',
+			style: ButtonStyle.Secondary,
+			customId: 'hit'
+		}),
+		new ButtonBuilder({
+			label: 'Stand',
+			style: ButtonStyle.Secondary,
+			customId: 'stand'
+		})
+	])
+];
+
+interface Card {
+	suit: string;
+	value: string;
+}
+
+// Define constants for suits and values
+const suits = ['Hearts', 'Diamonds', 'Clubs', 'Spades'];
+const values = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'Jack', 'Queen', 'King', 'Ace'];
+
+// Function to create a deck of cards
+function createDeck(): Card[] {
+	const deck: Card[] = [];
+	for (const suit of suits) {
+		for (const value of values) {
+			deck.push({ suit, value });
+		}
+	}
+	return deck;
+}
+
+// Function to shuffle the deck
+function shuffleDeck(deck: Card[]): void {
+	for (let i = deck.length - 1; i > 0; i--) {
```

Comment:
Needs to be i >=0 otherwise the loop where i = 0 is not executed, and you're just hoping the shuffle chooses card zero.

Basically if you don't do >=0, then the 0th card ends up unshuffled a disproportionate amount of the time.